src/util/transactionUtils.ts

```diff
@@ -14,8 +14,46 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
 
-import { TransactionStatusType, TransactionStatusEnum } from '../types/transactionTypes';
+import {
+  TransactionStatusType,
+  TransactionStatusEnum,
+  Payload,
+  TransactionDisplayPayload
+} from '../types/transactionTypes';
+import { encodeAddress } from '@polkadot/util-crypto';
 
 export function isTransactionCompleted(transaction: TransactionStatusType): boolean {
   return transaction.status === TransactionStatusEnum.COMPLETED;
 }
+
+interface PayloadInput {
+  payload: Payload;
+  ss58Format: number;
+  account: string;
+  createType: Function;
+  targetChain: string;
+}
+
+export function getTransactionDisplayPayload({
+  payload,
+  ss58Format,
+  account,
+  createType,
+  targetChain
+}: PayloadInput): TransactionDisplayPayload {
+  //@ts-ignore
+  const callType = createType(targetChain, 'BridgedOpaqueCall', payload.call);
+  //@ts-ignore
+  const call = createType(targetChain, 'Call', callType.toHex());
```

Comment:
```suggestion
  const call = createType(targetChain, 'Call', payload.call);
```
I think this should work just fine, no?

Reply:
i thought the same, but the api complained when it tried to built the type in that way.

Reply:
Maybe `payload.call` needs to be converted to hex first? I'm fine with this staying as-is, but it seems redundant ;)

Reply:
>  I'm fine with this staying as-is, but it seems redundant ;)

I know ! i believe the same as you but for some reason that i don't understand why it does not work. This is the error i get:
`Error: createType(Call):: findMetaCall: Unable to find Call with index 0x9490/[148,144]`